api/python/provisioner/commands/validator/node.py

```diff
@@ -0,0 +1,116 @@
+    data_netmask: str = NodeParams.data_netmask
+    data_private_interfaces: List = NodeParams.data_private_interfaces
+    data_private_ip: str = NodeParams.data_private_ip
+    data_public_interfaces: List = NodeParams.data_public_interfaces
+    data_public_ip: str = NodeParams.data_public_ip
+    hostname: str = NodeParams.hostname
+    mgmt_gateway: str = NodeParams.mgmt_gateway
+    mgmt_interfaces: List = NodeParams.mgmt_interfaces
+    mgmt_netmask: str = NodeParams.mgmt_netmask
+    mgmt_public_ip: str = NodeParams.mgmt_public_ip
+    roles: List = NodeParams.roles
+    cvg: List = NodeParams.cvg
+
+    _optional_param = [
+        'data_public_ip',
+        'roles',
+        'data_netmask',
+        'data_gateway',
+        'data_private_ip',
+        'mgmt_interfaces',
+        'mgmt_public_ip',
+        'mgmt_netmask',
+        'mgmt_gateway',
+        'cvg'
+    ]
+
+    def __attrs_post_init__(self):    # noqa: D105
+        params = attr.asdict(self)
+
+        # If storage.cvg.metadata or storage.cvg.data is specified,
+        # check entry for the other.
+        for data_set in params.get('cvg'):
+            logger.debug(f"DataSet being processed for CVG keys: {data_set}")
+            if (
+                data_set.get('data_devices') and
+                (
+                    (not data_set.get('metadata_devices')) or
+                    (data_set.get('metadata_devices') == UNCHANGED) or
+                    (data_set.get('metadata_devices') == '')
+                )
+            ):
+                raise ValueError(
+                    "List of data is specified. "
+                    "However, list of metadata is unspecified."
+                )
+            elif (
+                data_set.get('metadata_devices') and
+                (
+                    (not data_set.get('data_devices')) or
+                    (data_set.get('data_devices') == UNCHANGED) or
+                    (data_set.get('data_devices') == '')
+                )
+            ):
+                raise ValueError(
+                    "List of metadata is specified. "
+                    "However, list of data is unspecified."
+                )
+        if (
+            not 'physical' in GrainsGet().run(
+                'virtual',
+                targets=local_minion_id()
+            )[local_minion_id()]['virtual']
+        ):
+            self._optional_param.extend([
+                'bmc_user',
+                'bmc_secret'
+            ])
+
+        missing_params = []
+        for param, value in params.items():
+            if value == UNCHANGED and param not in self._optional_param:
+                missing_params.append(param)
+        if len(missing_params) > 0:
+            logger.error(
+                f"{missing_params} cannot be left empty or blank.. "
+                "These are mandatory to configure the setup."
+            )
+            raise ValueError(f"Mandatory param missing {missing_params}")
```

Comment:
There are no new line here after the last code line.

Reply:
@dmitrydb added !